c2rust-analyze/src/borrowck/mod.rs

```diff
@@ -64,6 +64,8 @@ pub enum OriginParam {
     /// A hypothesized region derived from a pointer type,
     /// e.g. `'h0` derived from the pointer in `*mut foo`
     Hypothetical(i64),
+    /// The 'static region
+    Static,
```

Comment:
`'static` doesn't make sense as an `OriginParam`.  `OriginParam` represents a lifetime parameter declaration (e.g. the `'a` in `struct S<'a> { ... }`), as opposed to `OriginArg` which is a "lifetime expression" used as an argument when referring to a generic type (e.g. the `'b` in `let x: S<'b> = ...;`).  It's not valid to write `'static` in a declaration position, such as `struct Foo<'static> { ... }`, so it shouldn't have an `OriginParam` variant.

Is `OriginParam::Static` actually used anywhere?  I don't see any mentions outside the `TryFrom` impl below.

Reply:
Yeah, I agree. This is because `Label` has `pub origin_params: &'tcx [(OriginParam, Origin)]` which is used in for e.g. `field_lty`. I guess this issue of `Static` is because of a deficiency in that implementation and it should instead potentially be `pub origin_args: &'tcx [(OriginArg, Origin)]`

Reply:
@spernsteiner I think I've addressed this issue in cde7b43e